tests/common/factories/annotation.py

```diff
@@ -77,15 +77,22 @@ def document_uri_dict():
         document_uri_dicts = [document_uri_dict()
                               for _ in range(random.randint(1, 3))]
 
-        def document_meta_dict(**kwargs):
+        def document_meta_dict(type_=None):
             """
             Return a randomly generated DocumentMeta dict for this annotation.
 
             This doesn't add anything to the database session yet.
             """
-            kwargs.setdefault('document', self.document)
-            kwargs.setdefault('claimant', self.target_uri)
+            kwargs = {
+                'document': None,
+                'claimant': self.target_uri,
+            }
+
+            if type_ is not None:
+                kwargs['type'] = type_
```

Comment:
I mixed in some code tidy up with the actual fix. See the individual commits for separate diffs